src/libs/Trie/index.js

```diff
@@ -0,0 +1,87 @@
+import _ from 'underscore';
+import TrieNode from './TrieNode';
+
+class Trie {
+    constructor() {
+        this.root = new TrieNode();
+    }
+
+    /**
+    * Add a word to the Trie
+    * @param {String} word
+    * @param {Object} [metaData]
+    * @param {TrieNode} node
+    * @param {String} [containChar]
+    * @returns {void}
+    */
+    add(word, metaData, node = this.root, containChar) {
+        if (word.length === 0 && !containChar) {
+            // throw new Error('Cannot insert empty word into Trie :', word);
+            throw new Error('Cannot insert empty word into Trie :', word);
+        }
+        if (word.length === 0) {
+            node.setCompleteWord();
+            node.setMetaData(metaData);
+            return;
+        } if (!node.children[word[0]]) {
```

Comment:
This contradicts with `EmojiTree`  because you are not passing `containChar`
<img width="509" alt="Screenshot 2022-08-16 at 1 21 02 AM" src="https://user-images.githubusercontent.com/85645967/184707228-f40222c2-4930-44dc-8476-cb41cbd585bb.png">

Reply:
@Santhosh-Sellavel when we call `add` as an object method we shouldn't pass `containChar` parameter, `containChar` is an indicator when it is `false` or `undefined` it indicates we are calling the `add` method for the first time otherwise indicates we are calling the `add` method from inside `add` body (as a recursive function) like here.
https://github.com/Karim-30/App/blob/b4352b843b523d37b580cf9f1bed821e57a3ae25/src/libs/Trie/index.js#L29-L31

we are using this to avoid adding an empty string because `add('')` means `word.length = 0` and `add('Hello')` will call `add('ello')` which call `add('llo')` until we reach `add('')` and then make it is a leaf node, the difference between both `add('')` is in the first one `containChar` is `undefined` so we should throw an error but in the second one `containChar` is 'o' so we should make it as a leaf node.

Reply:
You're right, sorry for the confusion.